test/integration/test_create_dss_docker_image.py

```diff
@@ -167,3 +190,104 @@ def test_docker_socket_on_host_touched(dss_container_context, fake_docker_socket
+        self._container = None
+
+    @contextmanager
+    def context(self):
+        with self._context_provider(self.socket_on_host) as container:
+            wait_for_socket_access(container)
+            self._container = container
+            yield self
+            self._container = None
+
+    def run(self, command: str, **kwargs) -> str:
+        return assert_exec_run(self._container, command, **kwargs)
+
+    def numeric_gid(self, group_entry: str) -> int:
+        """group_entry is "ubuntu:x:971:", for example"""
+        return int(group_entry.split(':')[2])
+
+    def get_gid(self, group_name: str) -> int:
+        output = self.run(f"getent group {group_name}")
+        return self.numeric_gid(output)
+
+    def get_unassigned_gid(self) -> int:
+        """
+        Return a new gid, that is not used for any other group, yet.
+        """
+        gid = 0
+        for line in self.run("cat /etc/group").splitlines():
+            if not line.startswith(f"nogroup:"):
+                gid = max(gid, self.numeric_gid(line))
+        return gid + 1
+
+    def assert_jupyter_member_of(self, group_name: str):
+        output = self.run(f"getent group {group_name}")
+        members = output.split(":")[3].split(",")
+        assert "jupyter" in members
+
+    def assert_write_to_socket(self):
+        signal = f"Is there anybody out there {datetime.now()}?"
+        self.run(
+            f'bash -c "echo {signal} > {DOCKER_SOCKET_CONTAINER}"',
+            user="jupyter")
+        assert signal == self.socket_on_host.read_text().strip()
+
+
+@pytest.fixture
+def socket_inspector(dss_container_context, accessible_file):
+    yield SocketInspector(dss_container_context, accessible_file)
+
+
+class GroupChanger:
+    def __init__(self, context_provider):
+        self._context_provider = context_provider
+
+    def chgrp(self, gid: int, path_on_host: Path):
+        path_in_container = "/mounted"
+        with self._context_provider(path_on_host, path_in_container) as container:
+            assert_exec_run(container, f"chgrp {gid} {path_in_container}")
+
+
+@pytest.fixture
+def group_changer(ubuntu_container_context):
+    return GroupChanger(ubuntu_container_context)
+
+
+def test_write_socket_known_gid(socket_inspector, group_changer):
+    with socket_inspector.context() as inspector:
+        gid = inspector.get_gid("ubuntu")
+
+    group_changer.chgrp(gid, socket_inspector.socket_on_host)
+
+    with socket_inspector.context() as inspector:
+        inspector.assert_jupyter_member_of("ubuntu")
+        inspector.assert_write_to_socket()
+
+
+def test_write_socket_unknown_gid(socket_inspector, group_changer):
+    with socket_inspector.context() as inspector:
+        gid = inspector.get_unassigned_gid()
+
+    group_changer.chgrp(gid, socket_inspector.socket_on_host)
+
+    with socket_inspector.context() as inspector:
+        assert gid == inspector.get_gid("docker")
+        inspector.assert_jupyter_member_of("docker")
```

Comment:
this the default case, also the entry point doesn't change anything, we need to additionally check, which gid the docker group has, it should be the selected unassigned one